frontend/model/contracts/group.js

```diff
@@ -301,13 +301,13 @@ function updateGroupStreaks ({ state, getters }) {
   }
 }
 
-const removeGroupChatroomProfile = (state, chatRoomID, member) => {
-  state.chatRooms[chatRoomID]['members'] =
+const removeGroupChatroomProfile = (state, chatRoomID, member, ourselvesLeaving) => {
+  state.chatRooms[chatRoomID].members =
     Object.fromEntries(
       Object.entries(state.chatRooms[chatRoomID].members)
         .map(([memberKey, profile]) => {
           if (memberKey === member && (profile: any)?.status === PROFILE_STATUS.ACTIVE) {
-            return [memberKey, { ...profile, status: PROFILE_STATUS.REMOVED }]
+            return [memberKey, { ...profile, status: ourselvesLeaving ? PROFILE_STATUS.LEFT_GROUP : PROFILE_STATUS.REMOVED }]
```

Comment:
Why is this function written in a such a bizarre way?

Wouldn't a straightforward implementation be a significantly more efficient O(1) 1-liner?

```js
state.chatRooms[chatRoomID].members[member].status = ourselvesLeaving ? PROFILE_STATUS.LEFT_GROUP : PROFILE_STATUS.REMOVED
```

Also, did you check all related code that checks the status for whether or not it handles `LEFT_GROUP` appropriately?

For example, below on line 322 we have a check for `status !== PROFILE_STATUS.REMOVED`, but what if the status is `LEFT_GROUP`?

Likewise, on 1262 we have another check:

```js
state.chatRooms?.[data.chatRoomID]?.members[memberID]?.status === PROFILE_STATUS.REMOVED
```